cmake/kobj.cmake

```
@@ -1,7 +1,14 @@
 # SPDX-License-Identifier: Apache-2.0
 
-set(GEN_KOBJ_LIST_SCRIPT ${ZEPHYR_BASE}/scripts/build/gen_kobject_list.py)
-set(GEN_KOBJ_LIST ${PYTHON_EXECUTABLE} ${GEN_KOBJ_LIST_SCRIPT})
+find_program(EXT_GEN_KOBJECT_LIST gen_kobject_list)
```

+1 for using `find_program()` as this allows users to make use of CMake built-in search features, like `CMAKE_PROGRAM_PATH` or by defining `GEN_KOBJ_LIST` directly.

But I think it's better to provide the names to look for, and allowing the variant without py extension will also make default substitution easier.

```suggestion
find_program(GEN_KOBJECT_LIST NAMES gen_kobject_list gen_kobject_list.py PATHS ${ZEPHYR_BASE}/scripts/build)
```

---

Oh yes, very neat!